src/parser/DependencyDictionary.h

```diff
@@ -151,5 +152,12 @@ class DependencyDictionary : public Dictionary {
   vector<vector<int> > maximum_right_distances_;
 };
 
+
+class DependencyTokenDictionary : public TokenDictionary {
+public:
+  DependencyTokenDictionary() {};
+  virtual ~DependencyTokenDictionary() {};
+  void InitializeFromDependencyReader(DependencyReader *reader);
```

Comment:
I think that now that you did this separation, these functions can be just called Initialize(...)